### ISSUE TITLE
WPM calculated based on the final words typed

### ISSUE THREAD

**Issue by gerazov** [NStefan002/speedtyper.nvim#17]
Currently the WPM is estimated as the number of characters typed divided by 5 as an average word length. This makes sense when typing words from a dictionary as they tend to be longer than the average word length. Also it nulls the variation of the WPM based on the specific words selected at the start of the game.

```lua
        wpm = (n_chars - n_mistakes) / 5 * (60 / time_sec)
        accuracy = (n_chars - n_mistakes) / n_keypresses * 100
```

On the other hand, when typing sentences, a more adequate calculation would be to take into account the actual content being typed. 

It makes sense to keep both approaches and let the user choose as an option.

Also, currently correcting mistakes in typing isn't taken into account. There can be arguments for and against this but it makes sense to also take this into account.

At the end the WPM and Accuracy should be calculated like so:
```lua
        wpm = n_words * (60 / time_sec)
        accuracy = n_correct / n_words * 100
```

**Comment by NStefan002** [comment]
> On the other hand, when typing sentences, a more adequate calculation would be to take into account the actual content being typed.

I agree. That's a way more precise calculation (especially for shorter game formats e.g. 15 seconds).

> Also, currently correcting mistakes in typing isn't taken into account. There can be arguments for and against this but it makes sense to also take this into account.

Corrections are counted in n_keypresses (they affect the accuracy, but not the wpm).
I was following a few discussions on a few different forums, and my general thought was that this was the most painless way to calculate wpm/accuracy (not so rough approximation, without considering too many things). That being said I totally agree with you, let the user decide.

I'll look into your PR in the next couple of days, sorry for the delay. And once again, thank you for your contributions to the project.

**Comment by gerazov** [comment]
Yeah sure - I'll test it out a bit more in the mean time :wink: 

One thing that occured to me is that it's better to take into account the `n_correct` for the wpm - it doesn't make sense to count wrong words towards the wpm. I'll fix that quick.

Another thing is that if the buffer gets refreshed it won't take into account the written lines before the clear ..

**Comment by gerazov** [comment]
Ok this is now fixed with the last commit.
